### ISSUE TITLE
[terra-clinical-result] Add support for checkbox in FlowsheetResultCell

### ISSUE THREAD

**Issue by shinepd** [cerner/terra-clinical#632]
# Feature Request

## Description
We'd like the ability to support checkboxes in the FlowsheetResultCell. Our use case is for device data that has not yet been authenticated in the system. This selection would operate independent of the current cell selection. This device data will not show a normalcy indicator, so we'd like it to be vertically aligned with the normalcy icon. 

It's technically possible for us to accomplish this on our own, though it would involve putting the checkbox on top of the cell and somehow referencing the padding for the normalcy icon to ensure the two are aligned.

Another option would be to allow consumers to supply a prop to suppress padding for the normalcy icon and have our checkbox take up that space. However, in this case we'd still need to know the padding for the normalcy as values above/below in the column may already be authenticated and show a normalcy icon.

## Additional Context / Screenshots
![Screen Shot 2020-03-20 at 8 47 03 AM](https://user-images.githubusercontent.com/61854307/77169510-96b82080-6a87-11ea-9103-b93eb8f1c805.png)

**Comment by neilpfeiffer** [comment]
Blocked by UX - before being made by Terra, open items need to be resolved:
- the checkmark box in place of the criticality icon is not part of a platform approved layout pattern for Terra, will need further validation or additional design decisions.
- a new visual design (possibly not a standard checkbox) may need to be created so the selectable cell UI is clearly separated/different from the combination of Clinical Result decorators (criticality icons, multiple results, and comment/modified/unverified icons).
- ui for selectability should instead be built-in to to a dedicated selectable cell in a flow-grid component, since the clinical-result>flowsheet-result-cell is purely for informational layout and does not hold a selectable state.

ON HOLD - Do not assign for work until updated design details have been provided.
